### PR TITLE
chore: move theme switch to full stack component

### PR DESCRIPTION
**Pull request by lsbyerley**
If merged, this PR moves the theme switch component out of root and into it's own full stack component.

I have made this change in my own codebase after coming across https://www.epicweb.dev/full-stack-components and in an effort to reduce bloat in the root file. If this change is not wanted, feel free to close :)

## Test Plan

Functionality and display of the theme switch component should not change.

## Checklist

- [ ] Tests updated
- [ ] Docs updated

### CODE REVIEW

File: app/routes/resources+/theme-switch.tsx [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+import { Icon } from '#app/components/ui/icon.tsx'
+import { useHints } from '#app/utils/client-hints.tsx'
+import { useRequestInfo } from '#app/utils/request-info.ts'
+import { type Theme, setTheme } from '#app/utils/theme.server.ts'
+
+const ThemeFormSchema = z.object({
+	theme: z.enum(['system', 'light', 'dark']),
+})
+
+export async function action({ request }: ActionFunctionArgs) {
```
Comment (kentcdodds) [inline app/routes/resources+/theme-switch.tsx:22]: This was copied over, but the original action in the root was not removed.

**Review by kentcdodds — CHANGES_REQUESTED** [review]
I'm a fan. Thanks. Just one thing.

**Comment by kentcdodds** [comment]
This will not negatively impact performance :)

File: app/routes/resources+/theme-switch.tsx (hunk 1 above)
Comment (lsbyerley) [reply app/routes/resources+/theme-switch.tsx:22]: ah good catch, thanks.. removed!

**Review by kentcdodds — APPROVED** [review]
Thanks!

**Pull request merged**